File: JavaScript/6js 1/allContinents.js
/*
|--------------------------------------------------------------------------
| Coding Meetup #8 - Higher-Order Functions Series - Will all continents be represented?
|--------------------------------------------------------------------------
|
*/

function allContinents(list) {
   const arr = ["Africa", "Americas", "Asia", "Europe", "Oceania"];
   const res = new Set();
   for (const dev of list) {
      res.add(dev.continent);
   }
   return arr.every((continent) => res.has(continent));
}


const allContinents = (arr) => new Set(arr.map((dev) => dev.continent)).size === 5;


function allContinents(list) {
   return ["Africa", "Americas", "Asia", "Europe", "Oceania"].every((x) =>
      list.some((y) => x == y.continent)
   );
}


let list2 = [
   {
      firstName: "Fatima",
      lastName: "A.",
      country: "Algeria",
      continent: "Africa",
      age: 25,
      language: "JavaScript",
   },
];
console.log(allContinents(list2), false);

File: Python/6py 1/all_continents.py
"""
|--------------------------------------------------------------------------
| Coding Meetup #8 - Higher-Order Functions Series - Will all continents be represented?
|--------------------------------------------------------------------------
|
"""

def all_continents(lst):
   continents = ["Africa", "Americas", "Asia", "Europe", "Oceania"]
   res = set(dev['continent'] for dev in lst)
   return all(continent in res for continent in continents)


def all_continents(lst): 
   return len(set(x["continent"] for x in lst)) == 5


def all_continents(lst): 
   return set(['Africa', 'Americas', 'Asia', 'Europe', 'Oceania']) == set([el['continent'] for el in lst])


list2 = [{'firstName': 'Fatima', 'lastName': 'A.', 'country': 'Algeria', 'continent': 'Africa', 'age': 25, 'language': 'JavaScript'}]
print(all_continents(list2), False)